modules/stochastic_tools/src/state_sim/model/actions/StateSimAction.C

```diff
@@ -0,0 +1,19 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "StateSimAction.h"
+#include "StateSimBase.h"
+#include <string>
+
+unsigned int StateSimAction::_counter = 0;
+
+StateSimAction::StateSimAction(std::string name, const ACTION_TYPE_ENUM & act_type)
+  : StateSimBase(name),
+    _act_type(act_type)
+{
+  _id = ++_counter;
```

Comment:
This probably won't behave correctly if we use this object in a multiapp setting.